include/realtime_tools/realtime_server_goal_handle.h

```diff
@@ -135,36 +146,41 @@ class RealtimeServerGoalHandle
 
   void runNonRealtime()
   {
-    if (valid())
+    if (!valid())
+      return;
```

Comment:
yea it looked to me that the loop originally was meant to spin even if there's no goal handle. it might have to be reworked by decoupling the function from it's object.

Reply:
If you could rule this out in this PR I would be happy. It seems to me as if that implementation is not clearly worked out in terms of usability. But that might just be my little knowledge about ros2 actions.

Reply:
https://github.com/ros-controls/ros2_controllers/issues/34